azuredevops/internal/acceptancetests/resource_resource_authorization_test.go

```diff
@@ -15,26 +15,65 @@ func TestAccResourceAuthorization_CRUD(t *testing.T) {
 	projectName := testutils.GenerateResourceName()
 	serviceEndpointName := testutils.GenerateResourceName()
 
-	resourcesHCL := testutils.HclServiceEndpointGitHubResource(projectName, serviceEndpointName)
+	serviceEndpointHCL := testutils.HclServiceEndpointGitHubResource(projectName, serviceEndpointName)
 	authedHCL := testutils.HclResourceAuthorization("azuredevops_serviceendpoint_github.serviceendpoint.id", true)
 	unAuthedHCL := testutils.HclResourceAuthorization("azuredevops_serviceendpoint_github.serviceendpoint.id", false)
 
+	tfAuthNode := "azuredevops_resource_authorization.auth"
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testutils.PreCheck(t, &[]string{"AZDO_GITHUB_SERVICE_CONNECTION_PAT"})
```

Comment:
I'm always thinking that we might have a check at the start of an AccTest, which skips the test if a required environment variable isn't set, instead of failing the test at all. The trade off could be that we don't detect misconfigured pipelines. The advantage would be that contributors could run AccTests more easily.

Reply:
I was thinking that too! But, I guess if we had a way to detech we were in a pipeline and then not skip them?

Reply:
Pipelines runners typically add some kind of environment variables to each job. 
[GitLab](https://docs.gitlab.com/ee/ci/variables/predefined_variables.html) has `CI`
[Travis](https://docs.travis-ci.com/user/environment-variables/#default-environment-variables) has `CI`
[Azure DevOps](https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml)  (Microsoft) is as always doing something else 😁 
GitHub had some discussions about it https://github.com/actions/runner/issues/368

But because people mainly use the `CI` environment variable, and you're always able to add environment variables to a  job or runners definition, I'd test for `CI=[true|1|yes|on]` (case insensitive regex of course 😀) to check if the code runs (or it's pretended) in a pipeline.

Reply:
Added ;)

Reply:
:grinning: cool, but I thought it should be implemented in the way that if CI is not set **and not** the required environment variable of the test, **than** it's skipped.

That ensures that  if we're running in a pipeline the test will fail if the environment variable of the test is not set but a developer can set the environment variable of the test and run the acceptance test outside a pipeline. :smiley:

Reply:
I don't follow?

Reply:
Okay  sorry ... used my mobile to answer back on your last comment and I had the impression you something different than changing `testutils.PreCheck`

I thought about an additional function like so 

```go
// SkipIfNotEnabled will skip a test if one of the environment variables are not set and the test is not running inside a pipelie (environment variable CI != "")
func SkipIfNotEnabled(t *testing.T, additionalEnvVars *[]string) {
	if os.Getenv("CI") == "" && additionalEnvVars != nil {
		for _, variable := range *additionalEnvVars {
			if strings.EqualFold(os.Getenv(variable), "") {
				t.Skipf("Skipping test %s, because environment variable is `%s` not set and required to run this test.", t.Name(), variable)
			}
		}
	}
}
```

And having a call to this function at the first line of an acceptance test

```go
func TestAccResourceAuthorization_CRUD(t *testing.T) {
	testutils.SkipIfNotEnabled(t, &[]string{"AZDO_GITHUB_SERVICE_CONNECTION_PAT"})

        ...
}
```

I'm a bit uncomfortable with your approach because firstly, it might skip **all** acceptance tests, including core tests that only require the Azure DevOps REST API (AZDO_PERSONAL_ACCESS_TOKEN) and need no external credentials or similar. 

And secondly, I am not a fan of extending (changing) the semantics of an existing function. Your modification to 'PreCheck' now causes tests to be skipped and the function did not do that before.

Reply:
Honestly, I don't think adding a new function call to all tests is a good solution (a lot of tests). Especially when you also have to reenter the env vars required.

I agree it should fail on basic on core tests. I can change it to fail on core tests or I can remove it completely.

Reply:
Maybe we should just bury the idea again. It also somehow collides with the Go build tags, which you can use, at least via the powershell script `acctest.ps1' with parameter `-tags', to skip certain acceptance tests. Not as convenient as the idea here, but also easy to use.

Reply:
So, I've gone and made the testing optional. I've placed it under a new make command and updated the docs